src/Gateways/Requests/ThreeDSecure/AbstractAuthenticationsRequest.php

```diff
@@ -35,4 +51,40 @@ protected function getToken( $requestData ) {
 
 		return $tokenResponse->paymentReference;
 	}
+
+	/**
+	 * Get (Billing) Address object from WC_Order.
+	 *
+	 * @return Address
+	 */
+	protected function get_wc_billing_address() {
+		$billingAddress = new Address();
+		$billingAddress->streetAddress1 = $this->order->get_billing_address_1();
+		$billingAddress->streetAddress2 = $this->order->get_billing_address_2();
+		$billingAddress->city           = $this->order->get_billing_city();
+		$billingAddress->state          = $this->order->get_billing_state();
+		$billingAddress->postalCode     = $this->order->get_billing_postcode();
+		$billingAddress->country        = $this->order->get_billing_country();
+		$billingAddress->countryCode    = CountryUtils::getNumericCodeByCountry( $billingAddress->country );
+
+		return $billingAddress;
+	}
+
+	/**
+	 * Get (Shipping) Address object from WC_Order.
+	 *
+	 * @return Address
+	 */
+	protected function get_wc_shipping_address() {
+		$shippingAddress = new Address();
+		$shippingAddress->streetAddress1 = $this->order->get_shipping_address_1();
+		$shippingAddress->streetAddress2 = $this->order->get_shipping_address_2();
+		$shippingAddress->city           = $this->order->get_shipping_city();
+		$shippingAddress->state          = $this->order->get_shipping_state();
+		$shippingAddress->postalCode     = $this->order->get_shipping_postcode();
+		$shippingAddress->country        = $this->order->get_shipping_country();
+		$shippingAddress->countryCode    = CountryUtils::getNumericCodeByCountry( $shippingAddress->country );
+
+		return $shippingAddress;
+	}
 }
```

Comment:
@apetrovici, can you add a new line in here, to be aligned with the others?
Everything else looks good.